### ISSUE TITLE
theme_update(text) fails due to NULL values

### ISSUE THREAD

**Issue by oneillkza** [hadley/ggplot2#1489]
It's possible, using `theme()`, to set individual font elements inline with plotting, eg (adapted from ggplot documentation example):

```{r}
df <- data.frame(gp = factor(rep(letters[1:3], each = 10)), y = rnorm(30))
ggplot(df, aes(x = gp, y = y)) + 
    geom_point() +
    theme(text=element_text(size=16))
```

However, when using theme_update to try to set this globally, it complains about `NULL` values in the `text` element.

```{r}
theme_update(text=element_text(size=16))
ggplot(df, aes(x = gp, y = y)) + 
    geom_point()
```

results in

```
Error in FUN(X[[i]], ...) : 
  Theme element 'text' has NULL property: family, face, colour, hjust, vjust, angle, lineheight, margin, debug
```

I don't know if this is intended, but it seems strange that the behaviour of `theme()` and `update_theme()` is different, and it's frustrating that it isn't possible to set font size globally.

(This is with `ggplot2_2.0.0`.)

**Comment by wmayner** [comment]
I'm having the same problem, though I'm updating the theme manually with 

```r
my.theme = theme_grey() + theme(text=element_text(family='Roboto', size=30))
```

As with OP, the same modification on a ``ggplot`` object works fine; it's only when trying to modify the default theme that this error is raised.

**Comment by oneillkza** [comment]
Interestingly, as a work-around, the following *does* work:

```{r}
theme_set(theme_bw(base_size = 16))
ggplot(df, aes(x = gp, y = y)) + 
    geom_point()
```

**Comment by hadley** [comment]
The base theme settings which are inherited by other settings need to be fully specified. Otherwise nothing gets inherited.

**Comment by oneillkza** [comment]
I see. So the solution for actually updating a single element of an element of a theme is something like:

```{r}
theme_update(text=(theme_grey()+theme(text = element_text(family = "Times")))$text )
```

That's pretty ugly; I'd expect theme_update to pull the base theme settings from the theme being updated, rather than requiring them to all be specified. It's your package, though.

**Comment by hadley** [comment]
@oneillkza there's probably a better way to do it than that, but I hardly ever use themes so I'm the wrong person to ask.

**Comment by oneillkza** [comment]
Thanks Hadley, that's understandable.

Hmm ... looking at the actual definition for `theme_update()`, it seems like the issue is that it uses `%+replace%` instead of `+`. 

```{r}
theme_update <- function(...) {
# Make a call to theme, then add to theme
theme_set(theme_get() %+replace% theme(...))
}
```

So for now the better way to update individual elements in a theme is to bypass `theme_update` and just to use `theme_set` directly:

```{r}
theme_set(theme_get() + theme(text=element_text(size=16)))
theme_get()$text
```

It'd be nice to have an update function that uses `+`, but I'll create a new issue articulating this, since this is becoming more of a feature request.

**Comment by oneillkza** [comment]
Or, rather, I'll make a pull request ;)

**Comment by eejd** [comment]
@oneillkza , Can't you just use the `theme()` for your needs?

**Comment by oneillkza** [comment]
@eejd In a round-about way (see conversation above), but anyway this problem was solved in #1519 , so I can just use the current version of the package ;)
